lib/android/app/src/main/java/com/reactnativenavigation/views/titlebar/TitleBar.java

```diff
@@ -101,26 +94,36 @@ public void setSubtitleFontSize(double size) {
     }
 
     public void setSubtitleAlignment(Alignment alignment) {
-        TextView subtitle = findSubtitleTextView();
-        if (subtitle == null) return;
-        alignTextView(alignment, subtitle);
+        subtitleAlignment = alignment;
     }
 
     private void alignTextView(Alignment alignment, TextView view) {
         Integer direction = view.getParent().getLayoutDirection();
         Boolean isRTL = direction == View.LAYOUT_DIRECTION_RTL;
-        int width = view.getResources().getDisplayMetrics().widthPixels;
-        view.post(() -> {
-            if (alignment == Alignment.Center) {
-                view.measure(0, 0);
-                //noinspection IntegerDivisionInFloatingPointContext
-                view.setX((width - view.getWidth()) / 2);
-            } else if (leftButtonController != null) {
-                view.setX(isRTL ? (getWidth() - view.getWidth()) - getContentInsetStartWithNavigation() : getContentInsetStartWithNavigation());
-            } else {
-                view.setX(isRTL ? (getWidth() - view.getWidth()) - UiUtils.dpToPx(getContext(), DEFAULT_LEFT_MARGIN) : UiUtils.dpToPx(getContext(), DEFAULT_LEFT_MARGIN));
-            }
-        });
+
+        if (alignment == Alignment.Center) {
+            //noinspection IntegerDivisionInFloatingPointContext
+            view.setX((getWidth() - view.getWidth()) / 2);
+        } else if (leftButtonController != null) {
+            view.setX(isRTL ? (getWidth() - view.getWidth()) - getContentInsetStartWithNavigation() : getContentInsetStartWithNavigation());
+        } else {
+            view.setX(isRTL ? (getWidth() - view.getWidth()) - UiUtils.dpToPx(getContext(), DEFAULT_LEFT_MARGIN) : UiUtils.dpToPx(getContext(), DEFAULT_LEFT_MARGIN));
+        }
+    }
+
+    @Override
+    protected void onLayout(boolean changed, int l, int t, int r, int b) {
+        super.onLayout(changed, l, t, r, b);
```

Comment:
Can we optimise  this so it realigns the title and subtitle only if `changed == true` ?